Info-Orbs/lib/globalTime/globalTime.cpp

```diff
@@ -93,12 +94,29 @@ String GlobalTime::getWeekday() {
     return m_weekday;
 }
 
+#include <HTTPClient.h> // Include the necessary header file
+
 bool GlobalTime::isPM() {
     return hour(m_unixEpoch) >= 12;
 }
 
-void GlobalTime::setTimeZoneOffset(int tz) {
-    Serial.print("Timezone Offset: ");
-    Serial.println(tz);
-    m_timeZoneOffset = tz;
-}
+void GlobalTime::getTimeZoneOffsetFromAPI() {
+    HTTPClient http;
+    http.begin(String(TIMEZONE_API_URL) + "?key=" + TIMEZONE_API_KEY + "&format=json&fields=gmtOffset&by=zone&zone=" + String(TIMEZONE_API_LOCATION));
+    int httpCode = http.GET();
+
+    if (httpCode > 0) {
+        JsonDocument doc;
+        DeserializationError error = deserializeJson(doc, http.getString());
+        if (!error) {
+            m_timeZoneOffset = doc["gmtOffset"].as<int>();
+            Serial.print("Timezone Offset from API: ");
+            Serial.println(m_timeZoneOffset);
+            m_timeClient->setTimeOffset(m_timeZoneOffset);
+        } else {
+            Serial.println("Failed to get timezone offset from API");
```

Comment:
the error should mention deserialization or something to be distinct from the http error

Reply:
yeah sure, cp mistake